classes/user/Repository.inc.php

```diff
@@ -393,4 +394,17 @@ public function getUserFromContextContact(Context $context): User
         $contextUser->setData('givenName', array_fill_keys($supportedLocales, $context->getData('contactName')));
         return $contextUser;
     }
+
+    /**
+     * Delete unvalidated expired users
+     * 
+     * @param object<Carbon\Carbon> $dateTillValid      The dateTime till before which user will consider expired
+     * @param array                 $excludableUsersId  The users id to exclude form delete operation
+     * 
+     * @return int The number rows affected by DB operation
+     */
+    public function deleteUnvalidatedExpiredUsers(Carbon $dateTillValid, array $excludableUsersId = []) {
```

Comment:
Maybe call `$dateTillValid` something like `$expiredRegistrationDate` or something.

Reply:
I also thought about `variable` name similar to `$expiredRegistrationDate` but changed it back to `$dateTillValid` as something like to `$expiredRegistrationDate` point out always something with registration date . But that method can be reused for other date purpose than only linked to registration date . So that way I think it's better to leave in such way so that is specify any `datetime` can be passed and it's meant to work that way .

Reply:
Ah ok, makes sense. It looks like the method is always checking against `date_registered`, though. Did you have plans to extend it for other things? I'm tripping up a bit on `$dateTillValid` because it's not clear what the date is meant to represent. I'd prefer to go with something that said what the date itself is:

```php
public function deleteUnvalidatedUsers(Carbon $registeredBefore)
```

Or something that said what the effect will be:


```php
public function deleteUnvalidatedUsers(Carbon $deleteUsersBefore)
```

But I'm probably bike-shedding at this point, so feel free to go with whatever you think is best! :smile: